python/pyspark/pandas/frame.py

```diff
@@ -11944,7 +11966,11 @@ def eval_func(pdf):  # type: ignore[no-untyped-def]
         if inplace:
             # Here, the result is always a frame because the error is thrown during schema inference
             # from pandas.
-            self._update_internal_frame(result._internal, requires_same_anchor=False)
+            # Since Spark 3.4, eval with inplace generates a new dataframe instead of operating
+            # in-place to follow pandas v1.4 behavior, see also SPARK-38946.
+            self._update_internal_frame(
+                result._internal, requires_same_anchor=False, anchor_force_disconnect=True
```

Comment:
ditto.

Reply:
See https://github.com/apache/spark/pull/36353#issuecomment-1178956945 , after confirm from panda community, we only add `setitem` make copy. so this line had been removed